mlflow/transformers/peft.py

```diff
@@ -0,0 +1,58 @@
+"""
+PEFT (Parameter-Efficient Fine-Tuning) is a library for efficiently adapting large pretrained
+models without fine-tuning all of model parameters but only a small number of (extra) parameters.
+Users can define a PEFT model that wraps a Transformer model to apply a thin adapter layer on
+top of the base model. The PEFT model provides almost the same APIs as the original model such
+as from_pretrained(), save_pretrained().
+"""
+_PEFT_ADAPTOR_DIR_NAME = "peft"
+
+
+def is_peft_model(model) -> bool:
+    try:
+        from peft import PeftModel
+    except ImportError:
+        return False
+
+    return isinstance(model, PeftModel)
+
+
+def get_peft_base_model(model):
+    """Extract the base model from a PEFT model."""
+    peft_config_map = model.peft_config or {}
+
+    # PEFT usually wraps the base model with two additional classes, one is PeftModel class
+    # and the other is the adaptor specific class, like LoraModel class, so the class hierarchy
+    # looks like PeftModel -> LoraModel -> BaseModel
+    # However, when the PEFT config is the one for "prompt learning", there is not adaptor class
+    # and the PeftModel class directly wraps the base model.
+    if peft_config := peft_config_map.get(model.active_adapter):
+        if not peft_config.is_prompt_learning:
```

Comment:
we can combine this if clause with the above line.

Reply:
unless it looks ugly.

Reply:
Yeah turns out the assignment expression doesn't add the variable to the scope of next condition, i.e. this doesn't work.
```
if peft_config := sth and peft_config.xxx
```
I can create `peft_config` outside the condition then it looks clearer with combined conditions.

Reply:
@B-Step62 I think `if peft_config := sth and peft_config.xxx` is equivalent to `if peft_config := (sth and peft_config.xxx)`. `if (peft_config := sth) and peft_config.xxx` should work.